### PR TITLE
Fixed #13212 - DbSession failure when trying to regenerate session id

### LINKED ISSUE #13212: Database exception using DbSession
### What steps will reproduce the problem?
When I'm trying to login on the website, from time to time I'm receiving an error message.
After reloading of the page everything is ok.
`YII_DEBUG `is enabled, `YII_ENV_DEV` is set to true.

Maybe this issue is linked with [this one](https://github.com/yiisoft/yii2/pull/1812)

[Stackoverflow question](http://stackoverflow.com/questions/41158812/yii2-dbsession-exception)
### What do you get instead?

> Integrity constraint violation – yii\db\IntegrityException
> 
> SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '0' for key 'PRIMARY'
> The SQL being executed was: UPDATE `session` SET `id`=0 WHERE `id`='3u5r7gq6pqh8s0gjrusfqpmni7'
> Error Info: Array
> (
>     [0] => 23000
>     [1] => 1062
>     [2] => Duplicate entry '0' for key 'PRIMARY'
> )
> ↵
> Caused by: PDOException
> 
> SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '0' for key 'PRIMARY'
> 
> in /var/www/develop/vendor/yiisoft/yii2/db/Command.php at line 844

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.10
| PHP version      | 7.0
| Operating system | Ubuntu 14.04

### PR DESCRIPTION
**Pull request by andrewnester**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #13212

### CODE REVIEW

**Review by samdark — APPROVED** [review]

File: framework/web/DbSession.php [hunk 1]
```diff
@@ -102,6 +102,10 @@ public function regenerateID($deleteOldSession = false)
 
         parent::regenerateID(false);
         $newID = session_id();
+        // if session id regeneration failed, no need to create/update it.
+        if (empty($newID)) {
+            return;
```
Comment (SilverFire) [inline framework/web/DbSession.php]: How about tracing `Yii::warning`?

**Review by SilverFire — CHANGES_REQUESTED** [review]

File: framework/web/DbSession.php (hunk 1 above)
Comment (samdark) [inline framework/web/DbSession.php]: Yes, good idea.

**Comment by andrewnester** [comment]
@SilverFire @samdark guys, I've just added `Yii::warning` call

**Review by SilverFire — APPROVED** [review]

**Comment by SilverFire** [comment]
Good to merge in 2.0.11

**Comment by samdark** [comment]
Thanks!

**Pull request merged**